clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/EventProcessor.java

```diff
@@ -16,120 +16,31 @@
-    }
+public interface EventProcessor<T> extends AutoCloseable {
 
     /**
-     * Drains all available events from the queue, and then processes them in order. If any errors are thrown while
-     * processing the individual events, these are submitted to the given {@link ProcessHandler}.
+     * Process an event that is received.
      */
-    protected boolean process(ProcessHandler<T> processHandler) {
-        closer.assertOpen("The processor was previously closed, so no further processing can occur");
-
-        List<T> events = drain();
-
-        if (events.isEmpty()) {
-            log.trace("No events to process");
-            return false;
-        }
+    void process(T event);
 
-        try {
-            log.trace("Starting processing of {} event{}", events.size(), events.size() == 1 ? "" : "s");
-
-            for (T event : events) {
-                try {
-                    Objects.requireNonNull(event, "Attempted to process a null event");
-                    log.trace("Processing event: {}", event);
-                    process(event);
-                    processHandler.onProcess(event, Optional.empty());
-                } catch (Throwable t) {
-                    KafkaException error = ConsumerUtils.maybeWrapAsKafkaException(t);
-                    processHandler.onProcess(event, Optional.of(error));
-                }
-            }
-        } finally {
-            log.trace("Completed processing");
-        }
-
-        return true;
-    }
-
-    /**
-     * It is possible for the consumer to close before complete processing all the events in the queue. In
-     * this case, we need to throw an exception to notify the user the consumer is closed.
-     */
-    private void closeInternal() {
-        log.trace("Closing event processor");
-        List<T> incompleteEvents = drain();
-
-        if (incompleteEvents.isEmpty())
-            return;
-
-        KafkaException exception = new KafkaException("The consumer is closed");
-
-        // Check each of the events and if it has a Future that is incomplete, complete it exceptionally.
-        incompleteEvents
-                .stream()
-                .filter(e -> e instanceof CompletableEvent)
-                .map(e -> ((CompletableEvent<?>) e).future())
-                .filter(f -> !f.isDone())
-                .forEach(f -> {
-                    log.debug("Completing {} with exception {}", f, exception.getMessage());
-                    f.completeExceptionally(exception);
-                });
-
-        log.debug("Discarding {} events because the consumer is closing", incompleteEvents.size());
-    }
-
-    /**
-     * Moves all the events from the queue to the returned list.
-     */
-    private List<T> drain() {
-        LinkedList<T> events = new LinkedList<>();
-        eventQueue.drainTo(events);
-        return events;
+    @Override
+    default void close() {
+        // Do nothing by default...
```

Comment:
Wouldn't it be safer to not have a default here as a reminder for closing.

Reply:
Yeah, I can see your point. TBH, I'm not sure if that's even needed still 🤔

Reply:
It turns out `close()` wasn't needed any more, so I just removed it 🤷‍♂️